module/zfs/vdev_raidz.c

```diff
@@ -1525,41 +1538,77 @@ vdev_raidz_io_start_write(zio_t *zio, raidz_row_t *rr, uint64_t ashift)
 {
 	vdev_t *vd = zio->io_vd;
 	raidz_map_t *rm = zio->io_vsd;
-	int c, i;
+	int c, i = 0, skip_first_cols = -1;
 
 	vdev_raidz_generate_parity_row(rm, rr);
 
-	for (int c = 0; c < rr->rr_cols; c++) {
+	IMPLY(rm->rm_nskip > 0, rm->rm_skipstart < rr->rr_scols);
```

Comment:
I have feeling that rm->rm_skipstart is always less then rr->rr_scols, unrelated to rm->rm_nskip.

Reply:
Agreed, I had a lot of this in there just for sanity checks while developing to make sure I did not make a mistake, but it can be removed. I removed it.